src/parquet/file/metadata.cc

```diff
@@ -368,7 +368,7 @@ class ColumnChunkMetaDataBuilder::ColumnChunkMetaDataBuilderImpl {
     column_chunk_->meta_data.__set_total_compressed_size(compressed_size);
     std::vector<format::Encoding::type> thrift_encodings;
     thrift_encodings.push_back(ToThrift(Encoding::RLE));
-    if (properties_->dictionary_enabled(column_->path())) {
+    if (has_dictionary) {
```

Comment:
The bug is that the column writer will not use dictionary encoding for `Boolean` type. We should depend on the column writer value and not on the default properties value.